src/public/modules/core/directives/on-click-outside.client.directive.js

```diff
@@ -0,0 +1,42 @@
+// Retrieved from https://codepen.io/robmazan/pen/gbeagj
+
+angular
+  .module('core')
+  .directive('onClickOutside', [
+    '$document',
+    '$parse',
+    '$timeout',
+    onClickOutside,
+  ])
+
+function onClickOutside($document, $parse, $timeout) {
```

Comment:
also could you explain to me why this directive is necessary vs `ng-blur`?

Reply:
It will screw up clicking on the dropdown options since blur will fire before the click, at least that was my experience, might have something to do with the layout also.

Also want to (not in this PR): have hover-activated dropdown close automatically onmouseleave, and focus-activated (i.e clicking on the avatar to activate) dropdown only dismissable with a click-outside. I realize this is not that accessible, I'll add a hidden screenreader-only button to close the dropdown in the dropdown options.

Reply:
I just tried this locally - why do you need the dropdown to open on hover rather than on click? Imho the fact that it opens on hover but doesn't disappear onmouseleave is kinda annoying, and even if it didn't open on hover, I'd click on it anyway when I see the red exclamation mark

Reply:
I'll update it according to my comments; this slipped my mind